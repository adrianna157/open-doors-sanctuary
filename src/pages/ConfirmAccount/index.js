import React, { useState, Fragment, useContext } from "react"
import { Auth, API, graphqlOperation } from "aws-amplify"
import { createUsers } from "../../graphql/mutations"
import HeaderText from '../../components/HeaderText'
import Button from '../../components/Button'
import TextField from '../../components/TextField'
import Container from '../../components/Container';
import Page from '../../components/Page';
import AuthHeader from "../../components/AuthHeader";
import AlertHandler from "../../components/AlertHandler";
import { AuthContext } from '../../helpers/AuthContext/AuthContext.js';
import { NotificationContext } from "../../helpers/AlertContext/AlertContext.js";

const ConfirmAccount = (props) => {
  const { authData } = props
  const [code, setCode] = useState("")
  const [state, dispatch] = useContext(AuthContext)
  const [alert, setAlert] = useContext(NotificationContext);

  const addUser = async (alias, uuid) => {
    try {
      var res = await API.graphql(graphqlOperation(createUsers, { input: {
        id: uuid,
        alias: alias,
        certification: "",
        heardAboutUs: ""
      } }))
      return res
    } catch (error) {
      setAlert({
        type: "SET_NOTIFICATION",
        payload: {
          occurs: true,
          message: error.message,
          textColor: "redText",
          borderColor: "redBorder",
        },
      });
    }
  }

  const signIn = () => {
    Auth.signIn({
      username: authData.email,
      password: authData.password,
    })
      .then((user) => {
        dispatch({type: 'SET_USER', payload: user})
        addUser(authData.alias, user.attributes.sub)
        authData.setAlias("")
        authData.setPassword("")
        props.onStateChange("signedIn", {})
      })
      .catch((err) => {
        setAlert({
          type: "SET_NOTIFICATION",
          payload: {
            occurs: true,
            message: err.message,
            textColor: "redText",
            borderColor: "redBorder",
          },
        });
      })
  }

  const confirmSignUp = () => {
    Auth.confirmSignUp(authData.email, code)
      .then((data) => {
        authData.setEmail("")
        setCode("")
        signIn()
      })
      .catch((error) => {
        setAlert({
          type: "SET_NOTIFICATION",
          payload: {
            occurs: true,
            message: error.message,
            textColor: "redText",
            borderColor: "redBorder",
          },
        });
      })
  }
  const resendCode = () => {
    Auth.resendSignUp(authData.email)
      .then(() => {
        console.log("code resent successfully")
      })
      .catch((error) => {
        setAlert({
          type: "SET_NOTIFICATION",
          payload: {
            occurs: true,
            message: error.message,
            textColor: "redText",
            borderColor: "redBorder",
          },
        });
      })
  }

  return props.authState === 'confirmSignUp' ? (
    <Page noLeftMargin>
      <AlertHandler></AlertHandler>
      <AuthHeader />
      <Container
        grayedBackground
        height="h-91pr lg:h-63pr"
        width="w-96pr lg:w-35pr"
        padding="pt-3pr px-4pr"
        className="flex flex-col"
        margin="mx-auto"
        mariginTop="mt-h8pr lg:mt-h26pr"
      >
        <center><HeaderText noBold fontSize="text-4xl" className="pb-2pr text-[#a26360]">Confirm Sign Up</HeaderText></center>
        <TextField
          type="code"
          value={code}
          onChange={(e) => setCode(e.target.value)}
          placeHolder="Code"
          label="Confirmation Code"
        />
        <Button solidBlue onClick={confirmSignUp} className="mt-4">
          Submit
        </Button>
        <div className="flex justify-end mt-4">
          <Button anchorTag onClick={resendCode}>
            Resend Code
          </Button>
        </div>
      </Container>
    </Page>
  ) : (<Fragment></Fragment>)
}

export default ConfirmAccount;
