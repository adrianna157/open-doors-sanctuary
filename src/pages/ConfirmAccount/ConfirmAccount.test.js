import React from 'react'
import Store from "../../helpers/AuthContext/AuthContext.js"
import AlertStore from "../../helpers/AlertContext/AlertContext.js";
import renderer from "react-test-renderer";
import ConfirmAccount from "./index.js";

describe('The CreateAccount Page Renders Correctly', () => {
    it('Should Render ConfirmAccount Page Component', () => {
        const component = renderer.create(
          <Store>
            <AlertStore>
              <ConfirmAccount authState="confirmSignUp" />
            </AlertStore>
          </Store>
        ).toJSON();
        expect(component).toMatchSnapshot()
    });

    it("Should not Render ConfirmAccount Page Component", () => {
      const component = renderer
        .create(
          <Store>
            <AlertStore>
              <ConfirmAccount />
            </AlertStore>
          </Store>
        )
        .toJSON();
      expect(component).toMatchSnapshot();
    });
});