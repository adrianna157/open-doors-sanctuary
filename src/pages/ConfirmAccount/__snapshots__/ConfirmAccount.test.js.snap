// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CreateAccount Page Renders Correctly Should Render ConfirmAccount Page Component 1`] = `
<div
  className="h-screen lg:h-auto lg:overflow-hidden  pb-6 mb-6"
>
  <div>
    <div
      className=" inset-y-h12pr -z-10 fixed  content-center flex-wrap h-46pr bg-no-repeat bg-cover bg-scroll w-screen bg-none  lg:flex lg:justify-center lg:bg-coin "
    />
    <div
      className="lg:flex lg:justify-center w-full"
    >
      <nav
        className="h-6pr bg-white fixed top-0 left-0 w-screen justify-center flex shadow-lg lg:h-12pr lg:justify-between lg:shadow-none"
      >
        <section
          className="h-4pr my-h08pr lg:flex lg:h-10pr lg:p-1"
        >
          <a
            href="/"
          >
            <img
              alt="Open Doors Sanctuary"
              className="h-4pr lg:h-10pr lg:ml-w05pr"
              src="guardian_group_logo.svg"
            />
          </a>
        </section>
      </nav>
      <div
        className="bg-white shadow-container lg:shadow-xl  w-96pr lg:w-35pr h-91pr lg:h-63pr mt-h8pr lg:mt-h26pr mx-auto rounded z-40 relative"
      >
        <div
          className="h-full standard-text-font container-shadow  pt-3pr px-4pr flex flex-col"
        >
          <center>
            <div
              className="mb-2 primary-blue-text font-display text-4xl pb-2pr"
            >
              Confirm Sign Up
            </div>
          </center>
          <div
            className="w-full"
          >
            <div
              className="flex"
            >
              <p
                className="light-primary-blue-text font-light mb-1"
              >
                Confirmation Code
              </p>

            </div>
            <input
              className="textfield-bg-color py-h3pr lg:py-h1pr w-full rounded h-4pr pl-4"
              onChange={[Function]}
              placeholder="Code"
              type="code"
              value=""
            />
          </div>
          <button
            className="btton primary-solid-blue mt-4"
            onClick={[Function]}
          >
            Submit
          </button>
          <div
            className="flex justify-end mt-4"
          >
            <button
              className="underline primary-blue-text"
              onClick={[Function]}
            >
              Resend Code
            </button>
          </div>
        </div>
      </div>
      <div
        className="container-grayed "
      />
    </div>
  </div>
</div>
`;

exports[`The CreateAccount Page Renders Correctly Should not Render ConfirmAccount Page Component 1`] = `null`;
